snap/validate_test.go

```diff
@@ -592,3 +592,184 @@ func (s *ValidateSuite) TestValidateSocketName(c *C) {
 		c.Assert(err, ErrorMatches, `invalid socket name: ".*"`)
 	}
 }
+
+func (s *YamlSuite) TestValidateAppStartupOrder(c *C) {
+	fooAfterBaz := []byte(`
+name: foo
+version: 1.0
```

Comment:
If the blurb above is irrelevant for the testing below, we can just type it out once inside the test itself and strip from all the cases.

Nice testing by the way, thanks!